airflow/providers/jenkins/operators/jenkins_job_trigger.py

```diff
@@ -160,7 +163,7 @@ def poll_job_in_queue(self, location, jenkins_server):
             location_answer = jenkins_request_with_headers(
                 jenkins_server, Request(method='POST', url=location))
             if location_answer is not None:
-                json_response = json.loads(location_answer['body'])
+                json_response = json.loads(location_answer['body'])  # type: ignore
```

Comment:
I couldn't make sense of this one:

```
incompatible type "Union[str, Mapping[str, str]]"; expected "Union[str, bytes, bytearray]"
                    json_response = json.loads(location_answer['body'])
                                               ^
```

I can't tell which one should become `str, bytes, bytearray`.

Reply:
Type of `location_answer` is `Union[str, Mapping[str, str]]` and this is visible in L54:
```
return {'body': response_body.decode('utf-8'), 'headers': response_headers}
```

What I would try is:
```suggestion
                body_str: str = location_answer['body']
                json_response = json.loads(body_str)
```

Reply:
I couldn't quite get this to work, but managed to pull something else together. See latest commit.